onnx/defs/nn/defs.cc

```diff
@@ -101,20 +101,12 @@ void convPoolShapeInference(
     if ((nullptr != auto_pad_attr) && (auto_pad_attr->s() != "VALID")) {
       int input_dims_size = static_cast<int>(n_input_dims);
       for (int i = 0; i < input_dims_size; ++i) {
-        int64_t residual = 0;
-        if (strides[i] > 1) {
-          if (!input_shape.dim(2 + i).has_dim_value()) {
-            continue;
-          }
-          residual =  input_shape.dim(2 + i).dim_value();
-          while (residual > 0) {
-            residual -= strides[i];
-          }
-        }
-        int64_t total_pad = residual == 0 ? kernel_shape[i] - strides[i] : kernel_shape[i] + residual;
-        if (total_pad < 0) {
```

Comment:
is(was) negative padding supported?

Reply:
Not sure if negative padding means anything in real cases. For above corner case, we can safely set total_pad to 0 if it's negative, but there might be more corner cases.

Reply:
Added a check if negative pads are present when explicitly provided. Since this is not supported, it will result in shape inference error.